cmd/osctl/cmd/cluster.go

```diff
@@ -118,238 +100,109 @@ func create() (err error) {
+			CIDR: *cidr,
+			MTU:  networkMTU,
+		},
+
+		Image:             nodeImage,
+		KubernetesVersion: kubernetesVersion,
 	}
 
 	// Create the master nodes.
-
-	requests := make([]*node.Request, masters)
-	for i := range requests {
-		requests[i] = &node.Request{
-			Input:    *input,
-			Image:    nodeImage,
-			Name:     fmt.Sprintf("%s-master-%d", clusterName, i+1),
-			IP:       net.ParseIP(ips[i]),
-			Memory:   memory,
-			NanoCPUs: nanoCPUs,
-		}
-
+	for i := 0; i < masters; i++ {
+		var typ generate.Type
 		if i == 0 {
-			requests[i].Type = generate.TypeInit
+			typ = generate.TypeInit
 		} else {
-			requests[i].Type = generate.TypeControlPlane
+			typ = generate.TypeControlPlane
 		}
-	}
 
-	if err := createNodes(requests); err != nil {
-		return err
+		request.Nodes = append(request.Nodes,
+			provision.NodeRequest{
+				Type:     typ,
+				Name:     fmt.Sprintf("%s-master-%d", clusterName, i+1),
+				IP:       net.ParseIP(ips[i]),
+				Memory:   memory,
+				NanoCPUs: nanoCPUs,
+			})
 	}
 
-	// Create the worker nodes.
-
-	requests = []*node.Request{}
-
 	for i := 1; i <= workers; i++ {
-		r := &node.Request{
-			Type:     generate.TypeJoin,
-			Input:    *input,
-			Image:    nodeImage,
-			Name:     fmt.Sprintf("%s-worker-%d", clusterName, i),
-			Memory:   memory,
-			NanoCPUs: nanoCPUs,
-		}
-		requests = append(requests, r)
+		request.Nodes = append(request.Nodes,
+			provision.NodeRequest{
+				Type:     generate.TypeJoin,
+				Name:     fmt.Sprintf("%s-worker-%d", clusterName, i),
+				Memory:   memory,
+				NanoCPUs: nanoCPUs,
+			})
 	}
 
-	if err := createNodes(requests); err != nil {
+	cluster, err := provisioner.Create(ctx, request)
+	if err != nil {
 		return err
 	}
 
 	// Create and save the osctl configuration file.
-
-	return saveConfig(input)
-}
-
-// nolint: gocyclo
-func createNodes(requests []*node.Request) (err error) {
-	var wg sync.WaitGroup
-
-	wg.Add(len(requests))
-
-	for _, req := range requests {
-		go func(req *node.Request) {
-			fmt.Println("creating node", req.Name)
-
-			if err = node.NewNode(clusterName, req); err != nil {
-				helpers.Fatalf("failed to create node: %w", err)
-			}
-
-			wg.Done()
-		}(req)
-	}
-
-	wg.Wait()
-
-	return nil
-}
-
-func destroy() error {
-	cli, err := client.NewEnvClient()
-	if err != nil {
+	if err = saveConfig(cluster); err != nil {
 		return err
 	}
 
-	filters := filters.NewArgs()
-	filters.Add("label", "talos.owned=true")
-	filters.Add("label", "talos.cluster.name="+clusterName)
-
-	containers, err := cli.ContainerList(context.Background(), types.ContainerListOptions{All: true, Filters: filters})
-	if err != nil {
-		return err
+	if !clusterWait {
+		return nil
 	}
 
-	var wg sync.WaitGroup
-
-	wg.Add(len(containers))
-
-	for _, container := range containers {
-		go func(container types.Container) {
-			fmt.Println("destroying node", container.Names[0][1:])
+	// Run cluster readiness checks
+	checkCtx, checkCtxCancel := context.WithTimeout(ctx, 20*time.Minute)
```

Comment:
this timeout should probably be a flag for `osctl cluster create`, it limits overall time for cluster readiness